hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -1995,9 +1996,10 @@ case class TableGroupWithinPartitions(child: TableIR, n: Int) extends TableIR {
     val prev = child.execute(ctx)
     val prevRVD = prev.rvd
     val prevRowType = prev.rvd.typ.rowType
-    val prevKeyType = prev.rvd.typ.kType
 
-    val rowType = prevKeyType ++ PCanonicalStruct(true, ("grouped_fields", PCanonicalArray(prevRowType, required = true)))
+    val rowType = PCanonicalStruct(required = true,
+      prev.rvd.typ.kType.fields.map(f => (f.name, f.typ)) ++
+        Array(("grouped_fields", PCanonicalArray(prevRowType, required = true))): _*)
```

Comment:
This can be just `("grouped_fields", PCanonicalArray(prevRowType, required = true))` right?

Reply:
er, how?

Reply:
I just meant:

```scala
val str = Seq("a", "b", "c") 
str ++ "D" 
// res1: Seq[Any] = List("a", "b", "c", 'D')
```

Reply:
vs:

```scala
str ++ Seq("D") 
res2: Seq[String] = List("a", "b", "c", "D")
```

Reply:
ah, I can use a single +. Note that what you've pasted above doesn't have the same result for top and bottom, though

Reply:
oh so strange, you're right. It's appending a character from "D". Thanks

Reply:
++ requires a traversable. String is implicitly convertible to Traversable[Char]. Sneaky scala